src/utils.rs

```diff
@@ -732,6 +833,112 @@ fn char_width(c: char) -> usize {
     }
 }
 
+/// Returns the number of visible characters, ignoring style related chars
+fn count_visible_chars(text: &str) -> usize {
```

Comment:
I'm pretty sure we already have API for this (`measure_text_width()`), and this fails to consider various Unicode things.

Reply:
It is indeed the case, it has been corrected.